src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,287 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import htsjdk.samtools.fastq.FastqReader;
+import htsjdk.samtools.fastq.FastqRecord;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import scala.Tuple2;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * To make this test runnable, "-DPath.To.SGA" should be set properly.
+ */
+// TODO: investigate why SGA, run with exactly the same parameters on the same machine, yields slightly different results: either SNP between the contigs or slightly different contig lengths
+// TODO: test if error messages are handled correctly
+public class RunSGAViaProcessBuilderOnSparkUnitTest extends CommandLineProgramTest {
+
+    private static final Path sgaPath = getSGAPath();
+    private static Path getSGAPath(){
+        String s = System.getProperty("Path.To.SGA");
+        if(null==s){
+            s = System.getenv("Path.To.SGA");
+        }
+        return (null==s) ? null : Paths.get(s);
+    }
+
+    private static final SGAModule indexer = new SGAModule("index");
+
+    // data block
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "spark/sv/RunSGAViaProcessBuilderOnSpark/");
+
+    private static final List<Tuple2<Long, URI>> rawFASTQFiles;
+    private static final Map<Long, URI> expectedAssembledFASTAFiles;
+    static {
```

Comment:
I think you should do this in a `@setup` method rather than a static initializer block.